src/btree/bt_sync_obsolete.c

```diff
@@ -11,6 +11,56 @@
 #define WT_CHECKPOINT_CLEANUP_FILE_INTERVAL 1 /* 1 second */
 #define WT_URI_FILE_PREFIX "file:"
 
+/*
+ * __sync_obsolete_limit_reached --
+ *     This function checks whether checkpoint cleanup can continue operating on the obsolete time
+ *     window pages.
+ */
+static bool
+__sync_obsolete_limit_reached(WT_SESSION_IMPL *session)
+{
+    WT_BTREE *btree;
+    WT_CONNECTION_IMPL *conn;
+
+    conn = S2C(session);
+    btree = S2BT(session);
+
+    if (__wt_atomic_load32(&btree->checkpoint_cleanup_obsolete_tw_pages) >=
+      conn->heuristic_controls.checkpoint_cleanup_obsolete_tw_pages_dirty_max)
+        return (true);
+    if (__wt_atomic_load32(&btree->eviction_obsolete_tw_pages) == 0 &&
+      __wt_atomic_load32(&btree->checkpoint_cleanup_obsolete_tw_pages) == 0 &&
+      __wt_atomic_load32(&conn->heuristic_controls.obsolete_tw_btree_count) >=
+        conn->heuristic_controls.obsolete_tw_btree_max)
+        return (true);
+    return (false);
+}
+
+/*
+ * __sync_obsolete_can_process --
+ *     TBD
+ */
+static bool
+__sync_obsolete_can_process(WT_SESSION_IMPL *session, WT_TIME_AGGREGATE ta)
```

Comment:
Name TBD

Reply:
how about `__sync_obsolete_tw_check`?

Reply:
Thank you! 9dea24e56d03f164908c2d87c43b349b5749e628